apps/www/registry/default/ui/breadcrumb.tsx

```diff
@@ -4,79 +4,73 @@ import { ChevronRight, MoreHorizontal } from "lucide-react"
 
 import { cn } from "@/lib/utils"
 
-const Breadcrumb = React.forwardRef<
-  HTMLElement,
-  React.ComponentPropsWithoutRef<"nav"> & {
+const Breadcrumb: React.FC<
+  React.ComponentProps<"nav"> & {
     separator?: React.ReactNode
   }
->(({ ...props }, ref) => <nav ref={ref} aria-label="breadcrumb" {...props} />)
+> = ({ ...props }) => <nav aria-label="breadcrumb" {...props} />
 Breadcrumb.displayName = "Breadcrumb"
 
-const BreadcrumbList = React.forwardRef<
-  HTMLOListElement,
-  React.ComponentPropsWithoutRef<"ol">
->(({ className, ...props }, ref) => (
+const BreadcrumbList: React.FC<React.ComponentProps<"ol">> = ({
+  className,
+  ...props
+}) => (
   <ol
-    ref={ref}
     className={cn(
-      "flex flex-wrap items-center gap-1.5 break-words text-sm text-muted-foreground sm:gap-2.5",
+      "text-muted-foreground flex flex-wrap items-center gap-1.5 break-words text-sm sm:gap-2.5",
       className
     )}
     {...props}
   />
-))
+)
 BreadcrumbList.displayName = "BreadcrumbList"
 
-const BreadcrumbItem = React.forwardRef<
-  HTMLLIElement,
-  React.ComponentPropsWithoutRef<"li">
->(({ className, ...props }, ref) => (
+const BreadcrumbItem: React.FC<React.ComponentProps<"li">> = ({
+  className,
+  ...props
+}) => (
   <li
-    ref={ref}
     className={cn("inline-flex items-center gap-1.5", className)}
     {...props}
   />
-))
+)
 BreadcrumbItem.displayName = "BreadcrumbItem"
 
-const BreadcrumbLink = React.forwardRef<
-  HTMLAnchorElement,
-  React.ComponentPropsWithoutRef<"a"> & {
+const BreadcrumbLink: React.FC<
+  React.ComponentProps<"a"> & {
     asChild?: boolean
   }
->(({ asChild, className, ...props }, ref) => {
+> = ({ asChild, className, ...props }) => {
   const Comp = asChild ? Slot : "a"
 
   return (
     <Comp
-      ref={ref}
-      className={cn("transition-colors hover:text-foreground", className)}
+      className={cn("hover:text-foreground transition-colors", className)}
```

Comment:
Here is an example of a place where I removed the `ref={ref}` line, as the `ref` is now included within `{...props}` as a `React.LegacyRef` type.